content/v1.11/getting-started/provider-aws-part-2.md

```diff
@@ -365,10 +360,16 @@ metadata:
 spec:
   group: custom-api.example.org
   names:
-    kind: database
-    plural: databases
+    kind: XDatabase
+    plural: xdatabases
 ```
 
+{{< hint "tip" >}}
+This new API endpoint is a _Composite Resource_ (XR) API endpoint. It is
+convention for Composite Resource kinds to start with `X`. For example
+`XDatabase`. This is only convention and not required.
+{{< /hint >}}
```

Comment:
It may make sense to move (and the claim equivalent) this into introduction.md.

Reply:
I know that there is a practice of using X as a preface to defining the resource but I found it confusing as I was getting started. It is obscuring the thing we're talking about.

I have a custom API endpoint and I'm creating my definition of a database from it. That is straight forward. In the XRD and Composition concept chapters we can expand on this and explain why X.

Reply:
Saying this convention is too confusing for the getting started guide but okay to introduce later doesn't sit well with me. If it's confusing it shouldn't exist. I'm also wary for the first experience most folks have with these concepts to differ from most other examples. All of Upbound's reference platforms use this pattern for example - https://marketplace.upbound.io/configurations/upbound/platform-ref-aws/v0.5.0.

Reply:
> I have a custom API endpoint and I'm creating my definition of a database from it. That is straight forward.

I'm not following this rationale at all, sorry.

Part of the reason we use the `X` convention is that, Crossplane concepts aside, the XR and the XRC represent the same thing. They're both API abstractions that represent "a database". "Database" and "CustomDatabase" doesn't make sense to me - one is not more custom than the other.

In an ideal world we might have both XR and XRC be of `kind: Database` with some other kind of differentiator, but Kubernetes makes that difficult. Early on we considered differentiating them by having the XR be `kind: Database` and the XRC be `kind: DatabaseClaim`. @bassam pointed out that that pattern is not ideal because the developers using this system probably just want to think about the claim as "my database", not "my database claim". The distinction between XRC and XR is more relevant to the platform engineers configuring Crossplane. So where we landed is that the claim gets `kind: Database` and the XR (which from the developer perspective is just "weird Crossplane machinery") gets the `kind: XDatabase` differentiator.

Reply:
I'm not a fan of the convention in general but I'm the new guy. I've struggled with understanding who has the `database` and who has the `Xdatabase`. I'd prefer to using more unique names or indicating who is creating the resource, like `compositeDatabase`

I don't see a problem changing the terms since a guide like this is about introducing the minimum number of steps/topics and not about best practices. The goal here is for a reader to understand each `kind` and what is' responsible for. From there we can introduce more complex topics and best practices or standards.

Regarding the reference platforms, they are great if you know Crossplane. If you're new it's very hard to connect the dots. I don't see a problem keeping them in the reference platforms.